File: CoinKeeper.BusinessLogic/Users/Models/Registration/RegisterModel.cs
using System.ComponentModel.DataAnnotations;

namespace Domain.Entities.Users.Registration;

public class RegisterModel
{
    public string Name { get; set; } = null!;

    [DataType(DataType.EmailAddress)]
    public string Email { get; set; } = null!;

    [DataType(DataType.Password)]
    public string Password { get; set; } = null!;

    [Compare("Password")]
    [DataType(DataType.Password)]
    public string ConfirmPassword { get; set; } = null!;
}

File: CoinKeeper.Api/ClientApp/src/models/User/RegisterModel.ts
export class RegisterModel {
    public Name: string = "";
    public Email: string = "";
    public Password: string = "";
    public ConfirmPassword: string = "";
    
    constructor(name: string, email: string, password: string, confirmPassword: string) {
        this.Name = name;
        this.Email = email;
        this.Password = password;
        this.ConfirmPassword = confirmPassword;
    }
}